projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -1,58 +1,133 @@
-        // Format values to be emmitted, this might change when implemented on the form
-        this._values = { ...this._values, [checkBoxVal]: !this._values[checkBoxVal] };
-        this.onChange.emit({ value: this._values });
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid())
+        };
+    }
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     *
+     * @memberof DotSelectComponent
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.status = getOriginalStatus(this.isValid());
+        this.emitValueChange();
+        this.emitStatusChange();
     }
 
     render() {
         return (
             <Fragment>
-                <label htmlFor={this._label}>{this.label}</label>
+                {getTagLabel(this.name, this.label)}
                 {this._options.map((item: DotOption) => {
-                    this._values = { ...this._values, [item.value]: this.value === item.value ? true : false };
+                    const trimmedValue = item.value.trim();
                     return (
-                        <div class='dot-checkbox__container'>
+                        <div class={getErrorClass(this.isValid())}>
                             <input
                                 type='checkbox'
-                                name={item.value}
-                                checked={this.value === item.value ? true : null}
+                                disabled={this.shouldBeDisabled()}
+                                id={this.value}
+                                checked={this.value.indexOf(trimmedValue) >= 0 || null}
                                 onInput={(event: Event) => this.setValue(event)}
-                                value={item.value}
+                                value={trimmedValue}
                             />
-                            <label htmlFor={item.value}>{item.label}</label>
+                            <label htmlFor={trimmedValue}>{item.label}</label>
                         </div>
                     );
                 })}
-                {this.hint ? <span class='dot-textfield__hint'>{this.hint}</span> : ''}
+                {getTagHint(this.hint)}
+                {getTagError(!this.isValid(), this.requiredmessage)}
             </Fragment>
         );
     }
+
+    // Todo: find how to set proper TYPE in TS
+    private setValue(event): void {
+        this.value = this.getValueFromCheckInputs(event.target.value.trim(), event.target.checked);
+        this.status = updateStatus(this.status, {
+            dotTouched: true,
+            dotPristine: false,
+            dotValid: this.isValid()
+        });
+        this.emitValueChange();
+        this.emitStatusChange();
+    }
+
+    private getValueFromCheckInputs(value: string, checked: boolean): string {
+        const valueArray = this.value.trim().length > 0 ? this.value.split(',') : [];
```

Comment:
You don't need `> 0`

Reply:
![image](https://user-images.githubusercontent.com/37185433/56754941-800c2b00-674b-11e9-9e44-23979ac202d2.png)

this is needed, otherwise it'll do a `split` when there's an empty string